### ISSUE TITLE
BertAdam's get_lr() not return correct learning rate

### ISSUE THREAD

**Issue by light8lee** [huggingface/pytorch-transformers#645]
when the initial values of `BertAdam`'s `params` have `requires_grad=False` Parameter, it will just continue the loop in `step()` function line 251, after step() function, when I want to use `get_lr()` to get the current learning rate, the state of this Parameter is a empty dict, so the function just return `[0]` in line 230. I think it is out of expectation, maybe more checking of the grad is needed somewhere.
